### PR TITLE
Many changes

### PR DESCRIPTION
**Pull request by DLochmelis33**
Summary:
* reorganize project structure to support JVM target
* implement a basic JVM runner
* a few extra refactors and renames
* convenience feature `AutoOutcome`
* integration with `@Test`

a) once again, this is a merge into `development`, not `main`
b) ignore random files that were "deleted", most of them were just moved and slightly changed (but enough for git to be unable to recognize that)

### CODE REVIEW

File: src/commonMain/kotlin/komem/litmus/AutoOutcome.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package komem.litmus
+
+interface AutoOutcome {
+    fun getOutcome(): Any?
+}
```
Comment (eupp) [inline src/commonMain/kotlin/komem/litmus/AutoOutcome.kt:5]: What is the purpose of `AutoOutcome` interface? 
What `Auto` prefix stands for?

File: src/commonMain/kotlin/komem/litmus/LTDefinition.kt [hunk 2]
```diff
@@ -30,11 +29,16 @@ class LTDefinitionScope<S>(
...
     fun build(): LTDefinition<S> {
-        if (!::outcomeFinalizer.isInitialized) throw IllegalStateException("outcome not set")
+        val outcomeFinalizer: S.() -> LTOutcome = when {
+            ::outcomeFinalizer.isInitialized -> outcomeFinalizer
+            stateProducer() is AutoOutcome -> {
+                { (this as AutoOutcome).getOutcome() }
+            }
+
+            else -> error("outcome not specified")
+        }
```
Comment (eupp) [inline src/commonMain/kotlin/komem/litmus/LTDefinition.kt:43]: What is the idea here? 
The use can inherit its data class from e.g. `IIOutcome` to treat fields `r1` and `r2` as outcomes automatically?

File: src/commonMain/kotlin/komem/litmus/LTDefinition.kt [hunk 3]
```diff
@@ -30,11 +29,16 @@ class LTDefinitionScope<S>(
     }
```
Comment (eupp) [inline src/commonMain/kotlin/komem/litmus/LTDefinition.kt:31]: Maybe add checks that `spec` and `outcome` are only called once during test configuration?

File: src/commonTest/kotlin/komem.litmus/infra/TestDefaults.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+expect val defaultParams: LTRunParams
+expect val defaultRunner: LTRunner
+
+fun LTDefinition<*>.run(
+    params: LTRunParams = defaultParams,
+    runner: LTRunner = defaultRunner,
+) {
+    val results = runner.runTest(params, this)
+    assertTrue { results.none { it.type == LTOutcomeType.FORBIDDEN } }
```
Comment (eupp) [inline src/commonTest/kotlin/komem.litmus/infra/TestDefaults.kt:17]: why isn't it a Kotlin/JUnit test?

File: src/jvmMain/kotlin/JvmMain.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import komem.litmus.barriers.JvmSpinBarrier
+import komem.litmus.testsuite.IRIWVolatile
+import kotlin.time.Duration.Companion.seconds
+
+fun main() {
+    val runner: LTRunner = JvmThreadRunner
+    val test = IRIWVolatile
+
+    val syncEverySchedule = generateSequence(10) { (it * 1.7).toInt() }.takeWhile { it < 1000 }.toList()
+    println("len = ${syncEverySchedule.size}")
```
Comment (eupp) [inline src/jvmMain/kotlin/JvmMain.kt:11]: What is this?
A comment explaining this logic would be helpful.

**Review by eupp — COMMENTED** [review]
Good job. 

A recommendation for the future: it is better to split independent features into different PRs, event if those are small. It is easier for review and conflict resolution. 

Also, please rebase on develop before merge.

File: src/commonMain/kotlin/komem/litmus/LTDefinition.kt (hunk 2 above)
Comment (DLochmelis33) [reply src/commonMain/kotlin/komem/litmus/LTDefinition.kt:43]: Yes, hence `AutoOutcome`

File: src/commonTest/kotlin/komem.litmus/infra/TestDefaults.kt (hunk 4 above)
Comment (DLochmelis33) [reply src/commonTest/kotlin/komem.litmus/infra/TestDefaults.kt:17]: This is a wrapper for writing actual tests, and it uses `kotlin.test.assertTrue`. What do you mean?

File: src/jvmMain/kotlin/JvmMain.kt (hunk 5 above)
Comment (DLochmelis33) [reply src/jvmMain/kotlin/JvmMain.kt:11]: This is a stub, I'm not sure what to put into `main()` for now

File: src/commonMain/kotlin/komem/litmus/LTDefinition.kt (hunk 3 above)
Comment (DLochmelis33) [reply src/commonMain/kotlin/komem/litmus/LTDefinition.kt:31]: Will do

On this topic: I wanted to put a DslMarker annotation here, but I couldn't figure out how. Does it require different receiver classes to work?

File: src/commonTest/kotlin/komem.litmus/infra/TestDefaults.kt (hunk 4 above)
Comment (eupp) [reply src/commonTest/kotlin/komem.litmus/infra/TestDefaults.kt:17]: Sorry, I thought this is a test on its own.

File: src/jvmMain/kotlin/JvmMain.kt (hunk 5 above)
Comment (eupp) [reply src/jvmMain/kotlin/JvmMain.kt:11]: I mean what is `syncEverySchedule` and how it is used?
It is not clear from the code.

File: src/commonMain/kotlin/komem/litmus/LTDefinition.kt (hunk 3 above)
Comment (eupp) [reply src/commonMain/kotlin/komem/litmus/LTDefinition.kt:31]: Could you point to your code with DslMarker?
I can have a look.

**Pull request merged**